elide-core/src/main/java/com/yahoo/elide/Elide.java

```diff
@@ -419,4 +449,41 @@ public RequestScope getRequestScope() {
             return requestScope;
         }
     }
+
+    private static void elideAcceptQuery(RequestScope requestScope, User opaqueUser,
+                                         Optional<MultivaluedMap<String, String>> queryParams, String path) {
+        if (requestScope.getElideSettings() != null) {
+            final QueryLogger ql = requestScope.getElideSettings().getQueryLogger();
+            if (opaqueUser == null) {
+                ql.acceptQuery(UUID.fromString(requestScope.getRequestId()),
+                        null,
```

Comment:
Instead of defining this same block twice, do the logic check on opaqueUser above and save the user principal (or null) in a local variable.